pkg/web3/sdk_v2.go

```diff
@@ -0,0 +1,225 @@
+type ContractsV2 struct {
+	LilypadPaymentEngine    *lilypadpaymentengine.LilypadPaymentEngine
+	LilypadProxy            *lilypadproxy.LilypadProxy
+	LilypadModuleDirectory  *lilypadmoduledirectory.LilypadModuleDirectory
+	LilypadContractRegistry *lilypadcontractregistry.LilypadContractRegistry
+	LilypadStorage          *lilypadstorage.LilypadStorage
+	LilypadUser             *lilypaduser.LilypadUser
+	LilypadTokenomics       *lilypadtokenomics.LilypadTokenomics
+	LilypadToken            *lilypadtoken.LilypadToken
+}
+
+type Web3SDKV2 struct {
+	Options      Web3Options
+	PrivateKey   *ecdsa.PrivateKey
+	Client       *ethclient.Client
+	CallOpts     *bind.CallOpts
+	TransactOpts *bind.TransactOpts
+	Contracts    *ContractsV2
+	Log          *zerolog.Logger
+}
+
+// Creates the new Go-binding wrapper modules for the new protocol
+func NewContractsV2(
+	options Web3Options,
+	client *ethclient.Client,
+	callOpts *bind.CallOpts,
+) (*ContractsV2, error) {
+	lilypadProxy, err := lilypadproxy.NewLilypadProxy(common.HexToAddress(options.LilypadProxyAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadModuleDirectory, err := lilypadmoduledirectory.NewLilypadModuleDirectory(common.HexToAddress(options.LilypadModuleDirectoryAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadPaymentEngine, err := lilypadpaymentengine.NewLilypadPaymentEngine(common.HexToAddress(options.LilypadPaymentEngineAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadContractRegistry, err := lilypadcontractregistry.NewLilypadContractRegistry(common.HexToAddress(options.LilypadContractRegistryAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadStorage, err := lilypadstorage.NewLilypadStorage(common.HexToAddress(options.LilypadStorageAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadUser, err := lilypaduser.NewLilypadUser(common.HexToAddress(options.LilypadUserAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadTokenomics, err := lilypadtokenomics.NewLilypadTokenomics(common.HexToAddress(options.LilypadTokenomicsAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadToken, err := lilypadtoken.NewLilypadToken(common.HexToAddress(options.LilypadL2TokenAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ContractsV2{
+		LilypadPaymentEngine:    lilypadPaymentEngine,
+		LilypadProxy:            lilypadProxy,
+		LilypadContractRegistry: lilypadContractRegistry,
+		LilypadModuleDirectory:  lilypadModuleDirectory,
+		LilypadStorage:          lilypadStorage,
+		LilypadToken:            lilypadToken,
+		LilypadUser:             lilypadUser,
+		LilypadTokenomics:       lilypadTokenomics,
+	}, nil
+}
+
+func NewContractSDKV2(ctx context.Context, options Web3Options, tracer trace.Tracer) (*Web3SDKV2, error) {
+	displayOpts := options
+	displayOpts.PrivateKey = "*********"
+	logger.Debug().Msgf("NewContractSDK: %+v", displayOpts)
```

Comment:
```suggestion
	logger.Debug().Any("options", displayOpts).Msg("Creating new contract SDK")
```
Zerolog has an `Any` that we may be able to use on the options.

Reply:
Yup, indeed seems to work:

![image](https://github.com/user-attachments/assets/eea563ef-34d2-4ea2-97e3-946c36548d8f)